### PR TITLE
Profile Picture Functionality

### LINKED ISSUE #237: FEATURE: Profile Picture Functionality
### Is there an existing issue raised for this?

- [X] I have searched the existing issues

### Describe the feature

- **Allow users to upload a profile picture during the signup process.**
- **Provide users with the ability to add or update their profile picture from the user dashboard.**

### Component

**This feature/component will be added on:**
- https://delightful-daifuku-a9f6ea.netlify.app/user/userHash
- https://delightful-daifuku-a9f6ea.netlify.app/signup

### It should be implemented because

- **It will enhance user experience**
- **It will create more personalized and engaging experience for users**
- **Moreover it will meet users expectations and will align with modern application standards** 

### Would you like to work on this issue?

Yes

### Terms & Conditions

- [X] I agree to follow the Code of Conduct
- [X] I'm a GSSOC'24 Ext contributor

### PR DESCRIPTION
**Pull request by k3uual**
## What does this PR do?

Fixes #237

## How should this be tested?

- Login into account and *click on edit button* provided besides profile picture
- Now click on any picture to *upload* from the file explorer that opened
- It will *update immediately on client* side and also *Base64* of the image will be *stored in Database*
- Now ***check Database*** (for Base64) and also try ***refreshing the page*** it should display correct image

https://github.com/user-attachments/assets/26572129-95ad-4c25-85e1-b1c1ca639d19

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read Contributing guide
- [x] Self-reviewed my own code
- [ ] Checked for warnings, there are none
- [x] Removed all `console.logs` (**Except `console.logs`  for *errors***)
- [x] The changes don't cause any responsiveness issues

### CODE REVIEW

**Comment by AlfiyaSiddique** [comment]
@k3uual please add an output video

File: backend/.env.example [hunk 1]
```diff
@@ -28,5 +28,8 @@ SMTP_PASSWORD=
 RESPONSE_MAIL=
 #huggingface api key for chatbot
 HUGGINGFACE_API_KEY=
+<<<<<<< HEAD
+=======
 #FRONT_END URL
-FRONT_END_URL=""
\ No newline at end of file
+FRONT_END_URL=""
+>>>>>>> master
```
Comment (AlfiyaSiddique) [inline backend/.env.example:35]: @k3uual remove these merge conflicts syntax  '>>>>'

File: backend/package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "type": "module",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "start": "node server.js",
+    "start": "nodemon server.js",
     "dev": "nodemon server.js"
```
Comment (AlfiyaSiddique) [inline backend/package.json:10]: @k3uual make it node instead of nodemon

File: backend/server.js [hunk 3]
```diff
@@ -17,8 +17,9 @@ app.use(express.urlencoded({ limit: "10mb", extended: true }));
 
 // CORS configuration
 const allowedOrigins = [
-  "https://delightful-daifuku-a9f6ea.netlify.app",
-  /https:\/\/deploy-preview-\d+--delightful-daifuku-a9f6ea\.netlify\.app/,
+    // "https://delightful-daifuku-a9f6ea.netlify.app",
+    // /https:\/\/deploy-preview-\d+--delightful-daifuku-a9f6ea\.netlify\.app/,
+    "http://localhost:5173"
 ];
```
Comment (AlfiyaSiddique) [inline backend/server.js:24]: @k3uual remove the local setup from here and put back the deploy urls

File: frontend/.env [hunk 4]
```diff
@@ -1,4 +1,10 @@
...
 # Add your google client ID here
-VITE_GOOGLE_CLIENT_ID=""
\ No newline at end of file
+<<<<<<< HEAD
+<<<<<<< HEAD
+VITE_GOOGLE_CLIENT_ID="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"
+# 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
+=======
+VITE_GOOGLE_CLIENT_ID=""
+>>>>>>> master
```
Comment (AlfiyaSiddique) [inline frontend/.env:10]: Same remove the unknown syntacs

File: frontend/common/backendUri.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+// const backendUri = "http://localhost:8080"
+
+// export default backendUri;
```
Comment (AlfiyaSiddique) [inline frontend/common/backendUri.js:3]: Why did you create this file when no code is in this? @k3uual

**Comment by k3uual** [comment]
> @k3uual some comments as well please fix them

Ok, will update all of it by today and also make a video showcasing the feature

**Comment by AlfiyaSiddique** [comment]
@k3uual looks great, please update the PR I will merge it

**Comment by k3uual** [comment]
> @k3uual looks great, please update the PR I will merge it

Yeah, I did update it and the latest commit was also automatically pushed in here (i mistakenly unassigned myself, can you please reassign me again)

**Comment by k3uual** [comment]
> @k3uual PR is not updated it also has some merge conflicts: image
> 
> Please pull, resolve and push

Ok will

> @k3uual PR is not updated it also has some merge conflicts: image
> 
> Please pull, resolve and push

Ok will do

**Comment by k3uual** [comment]
Fixed all the merge confilcts

**Review by AlfiyaSiddique — APPROVED** [review]

**Pull request merged**